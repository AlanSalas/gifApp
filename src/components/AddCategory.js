import { useState } from "react";
import PropTypes from "prop-types";
import { FcSearch } from "react-icons/fc";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form className="add-category" onSubmit={handleSubmit}>
      <div className="add-category__input">
        <FcSearch />
        <input
          type="text"
          placeholder="Search Gifs!"
          value={inputValue}
          onChange={handleChange}
        />
      </div>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
