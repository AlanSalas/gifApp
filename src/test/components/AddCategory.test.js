import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Tests in AddCategory component", () => {
  const setCategories = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Input should change his value", () => {
    const input = wrapper.find("input");
    const value = "spiderman";
    input.simulate("change", { target: { value } });
    expect(wrapper.find("p").text().trim()).toBe(value);
  });

  test("Dont submit if inputValue is empty", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Should submit and reset inputValue", () => {
    const input = wrapper.find("input");
    const value = "superman";
    input.simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop("value")).toBe("");
  });
});
