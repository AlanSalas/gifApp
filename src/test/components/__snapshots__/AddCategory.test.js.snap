// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in AddCategory component Should render properly 1`] = `
<form
  className="add-category"
  onSubmit={[Function]}
>
  <p />
  <div
    className="add-category__input"
  >
    <FcSearch />
    <input
      onChange={[Function]}
      placeholder="Search Gifs!"
      type="text"
      value=""
    />
  </div>
</form>
`;
